### ISSUE TITLE
Group by error

### ISSUE THREAD

**Issue by Svancara** [linq2db/linq2db#264]
Hello! Thanks for your hard work! 
I think I have found an error.
I am using MySql.

I have this LINQ:
```cs
var TABLE3 = from syst in C2SYSTEMS group syst by new { Y = syst.Added.Year };
```

where "Added" is DateTIme:
```cs
    public class c2ex_publicsystems
    {
        public c2ex_publicsystems();

        [Column]
        [NotNull]
        public DateTime Added { get; set; }

...snip....
    }
```

linq2db generates a wrong SQL command:
```sql
SELECT
    `t1`.`Added`
FROM
    `c2ex_publicsystems` `t1`
GROUP BY
    Extract(Year from `t1`.`Added`),
    `t1`.`Added`
```
Sincerely,
Bob

**Comment by Svancara** [comment]
Is this a solution? It works for me.

BasicSqlBuilder.cs:

protected virtual void BuildGroupByClause()
		{
			if (SelectQuery.GroupBy.Items.Count == 0)
				return;

var items = SelectQuery.GroupBy.Items.Where(i => !(i is SqlValue || i is SqlParameter)).ToList();

**// Bob: If there are SqlExpressions and the last one is SqlField, then remove SqlField:
            if (items.Count > 1 && items[items.Count - 2] is LinqToDB.SqlQuery.SqlExpression && items[items.Count - 1] is LinqToDB.SqlQuery.SqlField)
            {
                items.RemoveAt(items.Count - 1);
            }**

**Comment by sdanyliv** [comment]
@Svancara, I've  prepared correct fix for that issue #660

**Comment by MaceWindu** [comment]
Reopen, see #782
